### ISSUE TITLE
Middleware path-based mapping reapplies to PathBase on unhandled exception

### ISSUE THREAD

**Issue by AlexAlappsis** [aspnet/Tooling#443]
When adding a middleware mapping with IApplicationBuilder.Map(), the map path is reflected in HttpContext.Request.PathBase.  If the application experiences an unhandled exception, and the IApplicationBuilder isn't setup with an exception handler(i.e. UseDeveloperExceptionPage()) the PathBase will be appended to.

So if my map is "/v1", after an unhandled exception the HttpContext.Request.PathBase will be "/v1/v1"  Each unhandled exception will append the mapping.

**Comment by sayedihashimi** [comment]
@AlexAlappsis is this a problem with the templates that Visual Studio creates or something else?

**Comment by AlexAlappsis** [comment]
I discovered the problem working in a project that started as a Visual Studio template. I believe it's an issue with the middleware mapping functionality of the framework.  I can't be sure, but my guess is when there is a completely unhandled exception the application resets some things, including the middleware mapping, but the path base isn't being reset to a default before the mapping is reapplied.  I have made a simple test project that replicates the issue if you would like me put it somewhere.
